causalml/inference/tree/uplift.pyx

```diff
@@ -1298,8 +1327,8 @@ class UpliftTreeClassifier:
                     bestGain = gain
                     bestGainImp = gain_for_imp
                     bestAttribute = (col, value)
-                    best_set_left = [X_l, w_l, y_l]
-                    best_set_right = [X_r, w_r, y_r]
+                    best_set_left = [X_l, w_l, y_l, X_val_l, w_val_l, y_val_l]
+                    best_set_right = [X_r, w_r, y_r, X_val_r, w_val_r, y_val_r]
```

Comment:
`*_val_*` are available only when `X_val is not None`, and Ln #1330, 1331 should be conditioned on it. See the test log here: https://github.com/uber/causalml/actions/runs/5498114727/jobs/10028674273?pr=614#step:6:1317